react_main/src/pages/Play/Decks/CreateDeck.jsx

```diff
@@ -1,114 +1,324 @@
+        };
+      }
 
-  useEffect(() => {
-    if (params.get("edit")) {
       axios
-        .get(`/deck/${params.get("edit")}`)
+        .post(`/deck/profiles/create`, formData, {
+          headers: {
+            "Content-Type": "multipart/form-data",
+          },
+        })
         .then((res) => {
-          var deck = res.data;
+          siteInfo.clearCache();
+        })
+        .catch((e) => {
+          if (e.response == null || e.response.status == 413)
+            errorAlert("File too large, must be less than 2 MB.");
+          else errorAlert(e);
+        });
+    }
+  }
+
+  useEffect(() => {
+    document.title = "Create Anonymous Deck | UltiMafia";
+  }, []);
 
-          setEditing(true);
+  return (
+    <div className="deck">
+      <div className="main-section">
+        <div className="span-panel">
+          <form
+            onSubmit={handleSubmit((data) => {
+              onCreateDeck(editing, data);
+              siteInfo.clearCache();
+            })}
+          >
+            <h3>Deck Name</h3>
+            <input
+              className="deck-input"
+              type="text"
+              value={deckName}
+              onChange={(e) => setDeckName(e.target.value)}
+            />
+            {editing && (
+              <>
+                {fields.map((profile, index) => {
+                  return (
+                    <div className="inputs">
+                      <>
+                        <h4>Card #{index + 1}</h4>
+                        <section key={profile.id}>
+                          <ImageUpload
+                            indx={index}
+                            reg={register}
+                            setFile={setFile}
+                            watch={watch}
+                            getValues={getValues}
+                          ></ImageUpload>
+                          <label>
+                            <span>Name</span>
+                            <input
+                              {...register(`cards.${index}.name`)}
+                              defaultValue={profile.name}
+                            />
+                          </label>
+                          <label>
+                            <span>Color</span>
+                            <input
+                              className="color-input"
+                              type="color"
+                              {...register(`cards.${index}.color`)}
+                              defaultValue={profile.color}
+                            />
+                          </label>
+                          {/* <label>
+                    <span>Death Message</span>
+                    <input
+                      type="text"
+                      {...register(`cards.${index}.deathMessage`)}
+                      defaultValue={profile.deathMessage}
+                    />
+                  </label> */}
+                          <label>
+                            <span>Delete?</span>
```

Comment:
deleting profiles resets state to last save of deck
[](file:///home/joyce/Videos/Screencasts/Screencast%20from%2007-09-2023%2023:25:15.webm)